### PR TITLE
Remove abstraction layers from EventHubs WebJobs extensions

### PR DESCRIPTION
**Pull request by pakrym**
There are some layers in EventHubs WebJobs extensions that were the artifact of Track 2 refactoring. Remove them to simplify the code a bit.

Major changes:
* Merge `Processor` and `EventProcessorHost`
* Merge `ProcessorPartitionContext` and `Partition`, rename to `EventProcessorHostPartition`
* Remove ICheckpointer

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHost.cs [hunk 1]
```diff
@@ -13,155 +13,124 @@
 
 namespace Microsoft.Azure.WebJobs.EventHubs.Processor
 {
-    internal class EventProcessorHost
+    internal class EventProcessorHost : EventProcessor<EventProcessorHostPartition>
```
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHost.cs:16]: At some point, we probably want to consider renaming this, as `EventProcessorHost` specifically identifies a T1 type in the ecosystem and could be confusing.

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHost.cs [hunk 2]
```diff
@@ -13,155 +13,124 @@
...
-
-            internal virtual LeaseInfo GetLeaseInfo(string partitionId)
+            catch (Exception)
             {
-                if (LeaseInfos.TryGetValue(partitionId, out LeaseInfo lease)) {
-                    return lease;
-                }
-
-                return null;
+                // Best effort logging.
```
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHost.cs:93]: Is this a placeholder for the future or did we miss trimming the comment?

**Review by jsquire — APPROVED** [review]

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHost.cs (hunk 1 above)
Comment (pakrym) [reply sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHost.cs:16]: Yeah, there are so many things names *EventProcessor in the solution.

**Pull request merged**